### PR TITLE
[webui][api] Update nokogiri gem.

### PR DESCRIPTION
**Pull request by evanrolfe**
There is a security issue with libxml2 which is used by nokogiri.
(CVE-2017-0663).

### CODE REVIEW

File: src/api/.rubocop.yml [hunk 1]
```diff
@@ -141,6 +142,9 @@ Style/TernaryParentheses:
 Style/ZeroLengthPredicate:
   Enabled: true
 
+Style/FrozenStringLiteralComment:
```
Comment (Ana06) [inline src/api/.rubocop.yml]: why are you disabling this cop and how is this related to the update? :thinking:

**Comment by Ana06** [comment]
I wonder why is git-cop failing :thinking: :confused:

**Comment by Ana06** [comment]
Why does our test suite look different? I mean, linters are now separated, and where is git-cop?

**Comment by evanrolfe** [comment]
> Why does our test suite look different? I mean, linters are now separated, and where is git-cop? :confused: :confounded:

Because this is 2.8, before the linters ran first and before we had git-cop..

@bgeuken @mdeniz it seems that rubocop is checking files that were not changed in this PR which is preventing this from being merged. Can you please have a look at this? I'm not really sure what to do about it..

File: src/api/.rubocop.yml (hunk 1 above)
Comment (evanrolfe) [reply src/api/.rubocop.yml]: I have no idea whats going on with rubocop in this PR... It was telling me to add this comment to a file:
`# frozen_string_literal: true`

But that also made some specs fail too...

**Comment by Ana06** [comment]
@evanrolfe 

> Because this is 2.8, before the linters ran first and before we had git-cop..

right! :wink: 

@evanrolfe 

> @bgeuken @mdeniz it seems that rubocop is checking files that were not changed in this PR which is preventing this from being merged. Can you please have a look at this? I'm not really sure what to do about it..

From [Rubocop documentation](https://rubocop.readthedocs.io/en/latest/cops_style/#stylefrozenstringliteralcomment):

> This cop is designed to help upgrade to Ruby 3.0. It will add the comment  `# frozen_string_literal: true` to the top of files to enable frozen string literals. Frozen string literals will be default in Ruby 3.0. The comment will be added below a shebang and encoding comment. The frozen string literal comment is only valid in Ruby 2.3+.

So, I think the problem is that you aded [`TargetRubyVersion: 2.4`](https://github.com/openSUSE/open-build-service/pull/3902/files#diff-60d3df3c6f0836763f82698122e7ab47R4)

**Comment by evanrolfe** [comment]
> So, I think the problem is that you aded TargetRubyVersion: 2.4

Ok I removed TargetRubyVersion and now we get the original rubocop failure which caused me to add TargetRubyVersion in..

```
spec/controllers/webui/package_controller_spec.rb:344:20: E: unexpected token tLSHFT
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
        expected = <<~EOT
```

**Comment by Ana06** [comment]
There is a problem with the versions, because I think the Ruby version we are using to run Rubocop is not the same as the one run. That's why when you add `TargetRubyVersion: 2.4` the offenses change.

There is an issue in Rubocop: https://github.com/bbatsov/rubocop/issues/2956

This is closed, so it is supposed to be fixed, I'll write there or open a new issue because it seems it is not. We are not the only ones having problems with that, as if you check in the issue there are some people in the last months pointing this issue with similar issues.

So, until they fix it, I would stuck Rubocop version, as you did before but maybe taking 2.4.1 (`TargetRubyVersion: 2.4.1`) and we may want to correct the offenses instead of disable the cop or we will have to do it if at some point we want to update to Ruby 3.0. This cop support autocorrect, so you can just run:

`bundle exec rubocop --auto-correct --only FrozenStringLiteralComment`

File: src/api/.rubocop.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 inherit_from: .rubocop_todo.yml
 
 AllCops:
+  TargetRubyVersion: 2.4.1
```
Comment (Ana06) [inline src/api/.rubocop.yml]: @evanrolfe sorry, we are installing Ruby 2.4.0, so this should be `TargetRubyVersion: 2.4`

Can this and the regeneration of the todo file be in a different commit?

**Comment by bgeuken** [comment]
I don't get why we start having all these rubocops failing out of a sudden. I would prefer not to disable the whole rubocop test suite. Though after trying to disable the failing cop and having other cops fail instead I am okay to disable this for now.

**Comment by evanrolfe** [comment]
> DO NOT DISABLE THE WHOLE RUBOCOP! :rage: :sob: 

I don't see the difference is but OK!!!! :dagger: 

> I can make a PR that supersedes this want if you want @evanrolfe and we can merge that. :wink:

Yes, please do this! Thank you!

**Comment by Ana06** [comment]
@evanrolfe 

> > I can make a PR that supersedes this want if you want @evanrolfe and we can merge that. :wink:
> 
> Yes, please do this! Thank you!

Done! https://github.com/openSUSE/open-build-service/pull/3910 :wink:

**Pull request closed without merging**